menus/src/main/java/net/staticstudios/menus/parser/ButtonParser.java

```diff
@@ -0,0 +1,139 @@
+package net.staticstudios.menus.parser;
+
+import net.staticstudios.menus.action.ButtonAction;
+import net.staticstudios.menus.button.Button;
+import net.staticstudios.menus.button.PlayerSkullButtonBuilder;
+import net.staticstudios.menus.button.SimpleButtonBuilder;
+import org.bukkit.Material;
+import org.bukkit.configuration.ConfigurationSection;
+import org.slf4j.Logger;
+
+import java.util.*;
+
+public class ButtonParser {
+    private static final Set<String> VALID_ACTIONS = Set.of("close-menu", "go-back", "command", "send-message");
+
+    private final Logger logger;
+
+    public ButtonParser(Logger logger) {
+        this.logger = logger;
+    }
+
+    public boolean validateButton(String id, ConfigurationSection config) {
+        ConfigurationSection itemRepresentation = config.getConfigurationSection("item");
+
+        if (itemRepresentation == null) {
+            logger.warn("Invalid item representation! Def value not found in layout. {}.item", id);
+            return false;
+        }
+
+        int amount = config.getInt("item.amount", 1);
+        String itemName = config.getString("item.name");
+        String itemMaterial = config.getString("item.material");
+
+        if (itemName == null) {
+            logger.warn("Invalid item name! {}.item.name", id);
+            return false;
+        }
+
+        if (itemMaterial == null) {
+            String skullId = config.getString("item.head-id");
+            if (skullId == null) {
+                logger.warn("Invalid item material/head-id! {}.item.material, {}.item.head-id", id, id);
+                return false;
+            }
+        }
```

Comment:
Also the error messages could be simplified with `logger.warn("Invalid item material/head-id for {}", id)`
```suggestion
        String itemMaterial = config.getString("item.material", config.getString("item.head-id"));

        if (itemName == null) {
            logger.warn("Invalid item name! {}.item.name", id);
            return false;
        }

        if (itemMaterial == null) {
                logger.warn("Invalid item material/head-id! {}.item.material, {}.item.head-id", id, id);
                return false;
        }
```